### DISCUSSION TITLE
Typos LSP integration (low false-positive source code spell checker)

### DISCUSSION THREAD

**Discussion by jonathanslenders** [neovim/neovim#25976] [General]
There is this "typos" spell checker (https://github.com/crate-ci/typos) - extremely fast and written in Rust which now has an LSP server thanks to this project: https://github.com/tekumara/typos-vscode

I wonder if anyone has used this LSP server? I was trying to create a config for contributing to nvim-lspconfig, but unfortunately unsuccessful. Anyone interested to look into it?

The server config I have so far is this (right now for spell checking Python code):
```lua
local util = require 'lspconfig.util'

local root_files = {
  'pyproject.toml',
  'ruff.toml',
}

return {
  default_config = {
    cmd = { 'typos-lsp' },
    filetypes = { 'python' },
    root_dir = util.root_pattern(unpack(root_files)) or util.find_git_ancestor(),
    single_file_support = true,
    settings = {},
  },
  docs = {
    description = [[
    ...
    ]],
  },
}
```
It does spawn typos-lsp as a subprocess, but I'm not seeing any errors.

**Comment by jonathanslenders** [comment]
Actually, never mind. I got it to work: https://github.com/neovim/nvim-lspconfig/pull/2886

**Reply by fusillicode** [reply]
Thanks really a lot for your amazing work 🙇‍♂️ 

I was wondering if you ever tried to customize the [`diagnosticSeverity`](https://github.com/tekumara/typos-vscode#settings) of the server 🤔 

Sorry for bothering you here but I've [kind of tried everything](https://github.com/fusillicode/dotfiles/blob/f88964ee7334ba42c9fa0738dfa33a729d06f12c/nvim/lua/mason-tools.lua#L49-L63) (even compiled the LSP with some custom debugging) but I wasn't be able to change the `diagnosticSeverity` in any way 😞 

It seems that nothing is passed as `initializationOptions` from "Neovim" to `typos-lsp` 🤔

**Reply by fusillicode** [reply]
Ok never mind I was able to solve my issue and documented it [here](https://github.com/tekumara/typos-vscode/issues/17#issuecomment-1832355562) 😄 

Sorry once again for the bother 🙏
